### ISSUE TITLE
Curious bug

### ISSUE THREAD

**Issue by hellerve** [hellerve/zepto#4]
There is a curious bug that I discovered while extending the module system: using `let` inside a `load` inside a `define` does not work because the binding of `let` is lost on the way. Using a `let` inside a `define` or a `let` inside a `load` in itself does not yield an error.

**Comment by hellerve** [comment]
Issue still persists. I created a minimal working example:

``` clojure
((lambda () (macro-expand (list 'let '((x 1)) '(write x)))))))
; => (let ((x 1)) (write x))
(macro-expand ((lambda () (list 'let '((x 1)) '(write x))))))
; => ((lambda (x) (write x)) 1)
```

**Comment by hellerve** [comment]
Fixed in fa8ac814ef1d19a879e34de335b6be36a0985d5f.
